Add thing builder

lcc opened the pull request:
Add builder that facilitates the creation of the KNoT state machine.

Depends on #27

Commit:
Add publish data work for the publish data request

Commit:
Add builder for the KNoTThing

alla3 commented on knot-thing/src/main/java/com/cesar/knot_sdk/KNoTThingBuilder.kt:
```diff
@@ -0,0 +1,76 @@
+            hostname,
+            portNumber
+        )
+        //stMachine.knotMessageParser = KNoTMessageParser()
```
Why this line is commented? Uncomment it or remove this line.

alla3 commented on knot-thing/src/main/java/com/cesar/knot_sdk/KNoTThingBuilder.kt:
```diff
@@ -0,0 +1,76 @@
+    private var password = ""
+    private var thingName = ""
+
+
```
Remove a blank line.

Commit:
Add routing of received messages

Add routing of messages based on the binding key of the received
message.

Commit:
Add handler to start/stop messages timeouts

Commit:
Fix message from unregister event

Fix unregister event property to be of type DeviceRemoved.

This is necessary as the KNoTMessageUnregister is related to the
unregister request and not to the response.

Commit:
Add data request transtion

Commit:
Add data update event
